src/Proto.Actor/LocalContext.cs

```diff
@@ -124,18 +134,26 @@ public PID SpawnNamed(Props props, string name)
 
         public void SetBehavior(Receive receive)
         {
-            _behavior.Clear();            
+            _behavior = null;
             _receive = receive;
         }
 
         public void PushBehavior(Receive receive)
         {
+            if (_behavior == null)
+            {
+                _behavior = new Stack<Receive>();
+            }
             _behavior.Push(_receive);
             _receive = receive;
         }
 
         public void PopBehavior()
         {
+            if(_behavior == null)
+            {
+                return;

```

Comment:
Thinking about it, we should probably throw this ´throw new Exception("Can not unbecome actor base behavior");` when _behavior is null, as the meaning is the same as the later statements.